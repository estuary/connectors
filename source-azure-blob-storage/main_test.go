package main

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/assert"
)

const connectionString = "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;"

func TestAzureBlobStore_Read(t *testing.T) {
	// Create a test instance of the azureBlobStore
	cfg := config{
		Credentials: &credentials{
			ConnectionString: connectionString,
		},
		StorageAccountName: "devstoreaccount1",
		ContainerName:      "example",
	}
	az, err := newAzureBlobStore(cfg) // Pass cfg directly instead of its address
	assert.NoError(t, err)
	assert.NotNil(t, az)
}
func TestGetConfigSchema(t *testing.T) {
	parserJsonSchema := []byte(`{"type": "object", "properties": {"name": {"type": "string"}}}`)

	result := getConfigSchema(parserJsonSchema)

	snaps.WithConfig(
		snaps.Dir("snapshots"),
		snaps.Filename("config_schema"),
	).MatchJSON(t, string(result))
}
