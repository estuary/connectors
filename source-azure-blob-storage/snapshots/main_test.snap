
[TestGetConfigSchema - 1]
{
 "$schema": "http://json-schema.org/draft-07/schema#",
 "Properties": {
  "AzureClientID": {
   "description": "The client ID used to authenticate with Azure Blob Storage.",
   "order": 0,
   "secret": false,
   "title": "Azure Client ID",
   "type": "string"
  },
  "AzureClientSecret": {
   "description": "The client secret used to authenticate with Azure Blob Storage.",
   "order": 1,
   "secret": true,
   "title": "Azure Client Secret",
   "type": "string"
  },
  "AzureSubscriptionID": {
   "description": "The ID of the Azure subscription that contains the Azure Blob Storage account.",
   "order": 3,
   "secret": false,
   "title": "Azure Subscription ID",
   "type": "string"
  },
  "AzureTenantID": {
   "description": "The ID of the Azure tenant where the Azure Blob Storage account is located.",
   "order": 2,
   "secret": false,
   "title": "Azure Tenant ID",
   "type": "string"
  },
  "StorageAccountName": {
   "description": "The name of the Azure Blob Storage account.",
   "order": 4,
   "secret": false,
   "title": "Storage Account Name",
   "type": "string"
  },
  "matchKeys": {
   "description": "Filter applied to all object keys under the prefix. If provided, only objects whose absolute path matches this regex will be read. For example, you can use \".*\\.json\" to only capture json files.",
   "order": 5,
   "secret": false,
   "title": "Match Keys",
   "type": "string"
  },
  "parser": {
   "description": "",
   "order": 0,
   "secret": false,
   "title": "",
   "type": "object"
  }
 },
 "required": [
  "bucket",
  "region"
 ],
 "title": "S3 Source",
 "type": "object"
}
---
